src/helpers.jl

```diff
@@ -113,59 +113,73 @@ function override_pref(name, server_pref, override)
 
-        try
-            apis[Symbol(api_group(name))] = [KApi(api_group_type(pref_vers), api_typedefs(pref_vers))]
-        catch ex
-            if isa(ex, UndefVarError)
-                @info("unsupported $pref_vers")
-                continue
-            else
-                rethrow()
+            try
+                apis[Symbol(api_group(name))] = [KApi(api_group_type(pref_vers), api_typedefs(pref_vers))]
+            catch ex
+                if isa(ex, UndefVarError)
+                    @info("unsupported $pref_vers")
+                    continue
+                else
+                    rethrow()
+                end
             end
-        end
 
-        for api_vers in apigrp.versions
-            try
-                gt = api_group_type(api_vers.groupVersion)
-                td = api_typedefs(api_vers.groupVersion)
-                ka = KApi(gt, td)
-                kalist = apis[Symbol(api_group(name))]
-                (ka == kalist[1]) || push!(kalist, ka)
-            catch
-                @info("unsupported $(api_vers.groupVersion)")
+            for api_vers in apigrp.versions
+                try
+                    gt = api_group_type(api_vers.groupVersion)
+                    td = api_typedefs(api_vers.groupVersion)
+                    ka = KApi(gt, td)
+                    kalist = apis[Symbol(api_group(name))]
+                    (ka == kalist[1]) || push!(kalist, ka)
+                catch
+                    @info("unsupported $(api_vers.groupVersion)")
+                end
             end
         end
+    catch e
+        @retry if isa(e, IOError)
+            @debug("Retrying getAPIVersions")
+            sleep(2)
+        end
     end
     apis
 end
 
-function fetch_core_version(ctx::KuberContext; override=nothing, verbose::Bool=false)
+function fetch_core_version(ctx::KuberContext; override=nothing, verbose::Bool=false, num_tries=1)
     apis = ctx.apis
-    api_vers = getCoreAPIVersions(CoreApi(ctx.client))
-    name = "Core"
-    pref_vers = override_pref(name, api_vers.versions[1], override)
-    verbose && @info("Core versions", supported=join(api_vers.versions, ", "), preferred=pref_vers)
-    apis[:Core] = [KApi(getfield(@__MODULE__, Symbol("Core" * camel(pref_vers) * "Api")), getfield(getfield(@__MODULE__, :Typedefs), Symbol("Core" * camel(pref_vers))))]
-    for api_vers in api_vers.versions
-        try
-            gt = getfield(@__MODULE__, Symbol("Core" * camel(api_vers) * "Api"))
-            td = getfield(getfield(@__MODULE__, :Typedefs), Symbol("Core" * camel(api_vers)))
-            ka = KApi(gt, td)
-            kalist = apis[:Core]
-            (ka == kalist[1]) || push!(kalist, ka)
-        catch
-            @info("unsupported Core $api_vers")
+    @repeat num_tries try
+        api_vers = getCoreAPIVersions(CoreApi(ctx.client))
```

Comment:
wrapping just the `getCoreAPIVersions` should be enough, that's the only call with remote IO